cdp_scrapers/instances/portland.py

```diff
@@ -134,8 +139,65 @@ def get_matter(self, event_page: BeautifulSoup) -> Optional[Matter]:
         #   The first description phrase in each item on agenda page.
         #   This is usually hyperlinked to another page with more details.
 
+        # Find document number
+        doc_number_element_sibling = event_page.find(
+            "div", text=re.compile("Document number"), attrs={"class": "field__label"}
+        )
+
+        # If there is no document number, skip this minute item
+        if doc_number_element_sibling == None:
+            return None
+        doc_number_element = doc_number_element_sibling.next_sibling
+        doc_number = doc_number_element.find("div", class_="field__item").text.strip()
+
+        # Find title
+        title_div = event_page.find("div", class_="council-document__title")
+        matter_title = title_div.find("a").text.strip()
+
+        # Find type
+        title_div.find("a").clear()
+        matter_type = title_div.text.strip()
+        if matter_type[0] == "(" and matter_type[-1] == ")":
+            matter_type = matter_type[1:-1]
+
+        # Find result status
+        result_status_element_sibling = event_page.find(
+            "div", text=re.compile("Disposition"), attrs={"class": "field__label"}
+        )
+        result_status_element = result_status_element_sibling.next_sibling
+        result_status = result_status_element.text
+        if result_status in ["Accepted", "Passed"]:
+            result_status = MatterStatusDecision.ADOPTED
+        elif "Passed to" in result_status:
+            result_status = MatterStatusDecision.IN_PROGRESS
+        else:
+            result_status = None
+
+        # Find the sponsors
+        sponsor_element_uncle = event_page.find(
+            "div", text=re.compile("Introduced by"), attrs={"class": "field__label"}
+        )
+        sponsor_list = None
+        if not sponsor_element_uncle == None:
+            sponsor_element_parent = sponsor_element_uncle.next_sibling
+            sponsor_elements = sponsor_element_parent.findAll(
+                "div", class_="field__item"
+            )
+            sponsor_list = []
+            for sponsor_element in sponsor_elements:
+                sponsor = sponsor_element.text.strip()
+                sponsor_list.append(
+                    self.get_person(self.separate_name_from_title(sponsor))
+                )
+
         return self.get_none_if_empty(
-            Matter(matter_type=None, name=None, sponsors=None, title=None),
+            Matter(
+                matter_type=matter_type,
+                name=doc_number,
+                sponsors=sponsor_list,
+                title=matter_title,
+                result_status=result_status,
+            ),
         )
 
     def get_supporting_files(
```

Comment:
Shak I've already implemented this method in 1701bea45230139ca2bb0312015e716bfab7b300. Looks like your fork is not up-to-date.

Reply:
My first check-in yesterday was done before I merged. I then merged and did a second check-in. You can see your code in https://github.com/Shak2000/cdp-scrapers/blob/0a73ae634e480f3a981745f7aeb3cf6544afc296/cdp_scrapers/instances/portland.py#L203.

Reply:
Shak I'm referring to lines 214 and 226. You are skipping the entire method I wrote. Your code to look only for class_="council-document__title" works only for some files. There are other types of files that we must get in different ways. That is why wrote the method the way I did. If you think what I doesn't work that it should be fixed. but lines 214-226 just isn't the correct way to go about it, in my opinion.

Also, similar to `get_matter()` we need to change the method signature, since `get_event_minutes()` is passing in a `Tag` for a given minutes item, not the entire page `BeautifulSoup`.

Shak I think it's best to leave this method `get_supporting_files()` as-is without the lines you added. Also comment out line 376 where the method is called. Then I will update it to fit how it is being called from there after this PR is merged. e.g. Since you are passing in a `Tag` for a minutes item, I don't think we need the `minutes_item_index` parameter anymore.

Anyway I will confirm all those. Please revert out lines 214 - 226. I just don't think this is the best way to fix this method. What do you think?

Reply:
I hope I'm making sense. :sweat_smile: